app/uk/gov/hmrc/securemessage/services/Auditing.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.services
+
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+import uk.gov.hmrc.auth.core.Enrolments
+import uk.gov.hmrc.emailaddress.EmailAddress
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.EventTypes
+import uk.gov.hmrc.securemessage.controllers.model.ClientName
+import uk.gov.hmrc.securemessage.controllers.model.cdcm.write.CaseworkerMessage
+import uk.gov.hmrc.securemessage.controllers.model.common.write.CustomerMessage
+import uk.gov.hmrc.securemessage.models.core.Conversation
+import uk.gov.hmrc.securemessage.models.{ EmailRequest, QueryResponseWrapper }
+
+import scala.concurrent.ExecutionContext
+
+trait Auditing {
+
+  val auditConnector: AuditConnector
```

Comment:
This should be a `def` not a `val`, you can have out of order init issues with values and traits. https://stackoverflow.com/questions/19642053/when-to-use-val-or-def-in-scala-traits